### ISSUE TITLE
Failed to scale down the self hosted etcd clusters

### ISSUE THREAD

**Issue by yifan-gu** [coreos/etcd-operator#934]
The pluton test for bootkube is trying to scale down from 3 to 1, and it often gets stuck.
Here is some artifacts I got:

**etcd cluster TPR object**:
```
$ kubectl get clusters -n kube-system -o yaml
apiVersion: v1
items:
- apiVersion: etcd.coreos.com/v1beta1
  kind: Cluster
  metadata:
    creationTimestamp: 2017-04-04T00:28:46Z
    name: kube-etcd
    namespace: kube-system
    resourceVersion: "1785"
    selfLink: /apis/etcd.coreos.com/v1beta1/namespaces/kube-system/clusters/kube-etcd
    uid: aa4ecb7d-18cd-11e7-9d37-42010af0000e
  spec:
    TLS: null
    pod:
      antiAffinity: false
      nodeSelector:
        master: "true"
      resources: {}
    selfHosted:
      bootMemberClientEndpoint: http://10.240.0.14:12379
    size: 1
    version: 3.1.0
  status:
    conditions:
    - reason: 'Current cluster size: 0, desired cluster size: 1'
      transitionTime: 2017-04-04T00:28:46.309813657Z
      type: ScalingUp
    - reason: ""
      transitionTime: 2017-04-04T00:30:47.275086856Z
      type: Ready
    - reason: 'Current cluster size: 1, desired cluster size: 3'
      transitionTime: 2017-04-04T00:33:42.513059964Z
      type: ScalingUp
    - reason: 'Current cluster size: 2, desired cluster size: 3'
      transitionTime: 2017-04-04T00:36:00.546899788Z
      type: ScalingUp
    controlPaused: false
    currentVersion: 3.1.0
    members:
      ready:
      - kube-etcd-0000
      - kube-etcd-0001
    phase: Running
    reason: ""
    size: 3
    targetVersion: ""
kind: List
metadata: {}
resourceVersion: ""
selfLink: ""
```

**etcd operator logs**: [here](https://gist.github.com/yifan-gu/836b47963c2665bf57cf9d17e3841ce9)

**kube-etcd logs**: [here](https://gist.github.com/yifan-gu/f6bda8a120082b9e5c014be3f280f856)

@xiang90 @hongchaodeng 

I also keep the cluster running, so you can ssh into them and do more debug.

```shell
gcloud compute ssh yifan-bootkube-test-16f15d65500a0db42e11 --zone us-central1-a
cd /home/core
export KUBECONFIG=/etc/kubernetes/kubeconfig
sudo ./kubectl get nodes
```

**Comment by hongchaodeng** [comment]
@yifan-gu 
```
$ gcloud compute ssh yifan-bootkube-test-16f15d65500a0db42e11 --zone us-central1-a

ERROR: (gcloud.compute.ssh) Could not fetch instance:
 - The resource 'projects/coreos-k8s-scale-testing/zones/us-central1-a/instances/yifan-bootkube-test-16f15d65500a0db42e11' was not found
```

**Comment by yifan-gu** [comment]
@hongchaodeng Oh it's in the `coreos-gce-testing` project.
```
gcloud config set project coreos-gce-testing
```

**Comment by hongchaodeng** [comment]
pod `kube-etcd-0002` exited before it's removed by operator.

**Comment by hongchaodeng** [comment]
pod `kube-etcd-0002` logs:
```
2017-04-04 00:37:41.908054 E | etcdserver: the member has been permanently removed from the cluster
2017-04-04 00:37:41.908145 I | etcdserver: the data-dir used by this member must be removed.
```

It seemed like there is previous data dir left. 
etcd pod use hostPath. The data dir should be cleaned up after each test.

**Comment by yifan-gu** [comment]
Any idea why that happens?
The pluton test code is [here](https://github.com/coreos/mantle/blob/master/pluton/tests/bootkube/scale.go)
Seems it does a scale up and then a scale down.

**Comment by hongchaodeng** [comment]
@yifan-gu 

When etcd pod starts up, it uses hostpath `/var/etcd/${namespace}-${cluster-name}`.

I don't know the details on what happened. You can check if the data dir exists before the test. It could be left by other tests.

**Comment by yifan-gu** [comment]
@hongchaodeng Hmm, I only ran 1 single test, so it doesn't seems like this is "left be other tests". Besides, even running tests in parallel, the pluton tests are running in a separate cluster. I could take a look tomorrow.

**Comment by hongchaodeng** [comment]
@yifan-gu 
Any update?

**Comment by yifan-gu** [comment]
@hongchaodeng nope

**Comment by hongchaodeng** [comment]
I'm closing this since this isn't an issue anymore.
Feel free to reopen it if you still encounter problems

**Comment by yifan-gu** [comment]
> I'm closing this since this isn't an issue anymore.

I don't feel the issue is solved yet, cc @pbx0 if you can provide more info for the selfhosted etcd scale test flakiness.

**Comment by hongchaodeng** [comment]
@yifan-gu 
If it happened again, let's create a new issue in bootkube or pluton with more logs and detailed repro steps.
